### ISSUE TITLE
Migration UI - 1 - Home page 

### ISSUE THREAD

**Issue by exezbcz** [kodadot/nft-gallery#7560]
part of a bigger issue:
- #7552 

# Home page

## Top part
<img width="785" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/6c93f71d-910b-47b6-b8d8-40f89f6e4a83">
- introduces into migration, shows important parts - like the route etc.

### source and destination
- same as in teleport v2 (#7363) 
- gives the option to choose whether migrating from Polkadot assethub or Kusama asset hub. The default is polkadot. 
<img width="350" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/00c57ad8-8a02-4788-8c49-ced4350ed7a7">
- The source is Kusama, or if its more clearer maybe rmrk2, It is up to the devs 

### Faq
- I have decided to put faq on the top there
- migration is pretty hard to understand for new users. This will hopefully ensure the users are not confused about what is happening. 
<img width="428" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/fe4eb84f-046d-400d-9a97-289cf997faeb">

- states are on the left: 
<img width="457" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/7f9a04a5-6e2f-4d0f-8cc9-8e9369914074">
- please note that the height will be taken from the card with the most text - other cards will inherit the height (that means cards are consistent and won't jump around) 
- if you would like to add anything as an explainer to the cards, feel free to comment. 

## Selection
- under the full-width divider, there is a contextual message with the account you are connected with, and then there is a little tooltip section. 
- there is also a state for when you are not connected. 
<img width="838" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/139d449d-dd57-4bd5-8351-a2f7e0bd3c78">
- tooltip copy for that as well. 
<img width="370" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/eb3aea1a-ae74-43b5-aada-0d82d65f257b">

- the selection is filled with drop landing page-like cards. 
<img width="489" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/1c412ccb-cfbb-4dec-83af-3f0b4c9419a5">

- cards have a hover effect, and the migrate button itself has one as well. 
<img width="291" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/2735cd73-a36c-45d4-857c-f7da7b605c90">

- you can try it yourself in the prototype next to the card component. 

### Ready for Migration
- Here, you will be able to view collections that you own.
- >As the creator or owner of a collection, you can initiate migration here. Keep in mind only items owned by you can be migrated at this stage. You'll need to pre-sign items owned by others to allow individual migrations.
- if there is no collection to be migrated from this part, this part is simply hidden. 

### Waiting for your action
<img width="512" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/59b9507c-f164-4d6b-b3e5-c5581d02bbe3">

- here are items from collections you don't own, but the collection was migrated and initiated - giving you the option to migrate as well. 
- >Displayed here are individual items that you own from collections that have been migrated. Ensure you're up-to-date by migrating these pieces right away. Please note you can only start the migration once the collection owner initiates it.

### migration not possible
- pretty sure there will be some collections that you won't be able to migrate. For this we have the last part under divider. 
<img width="714" alt="image" src="https://github.com/kodadot/nft-gallery/assets/90852205/d45afd8d-0138-45a9-8b8f-d58a6b980b9f">

**Comment by yangwao** [comment]
Hey @kodadot/internal-dev if someone would pick this one to start?

**Comment by preschian** [comment]
I will check this one

**Comment by preschian** [comment]
Actually, I'm still not familiar with the migration script. I will try to finish the UI first with dummy data and integrate it with real data later

**Comment by kodabot** [comment]
ASSIGNED - @preschian 🔒 LOCKED -> Saturday, October 14th 2023, 22:48:15 UTC -> 36 hours

**Comment by vikiival** [comment]
> Actually, I'm still not familiar with the migration script. I will try to finish the UI first with dummy data and integrate it with real data later

I think let's start with UI.

**Comment by kodabot** [comment]
ASSIGNMENT EXPIRED - @preschian has been unassigned.

**Comment by preschian** [comment]
@vikiival how to check if the collection is ready to migrate, waiting, or not supported?

**Comment by vikiival** [comment]
Any collection issuer and current owner of is possible to be migrated
On both KSM and RMRK

**Comment by preschian** [comment]
![image](https://github.com/kodadot/nft-gallery/assets/734428/240392eb-ebb9-4747-a4dc-3fbfa959cb3f)

@vikiival 
1. How do we check if the collection has already been migrated?
2. Does it mean migration was impossible if the collection owner had yet to migrate? Is that correct?

**Comment by vikiival** [comment]
> How do we check if the collection has already been migrated?

Good Q.
Will add it to the API.
The full desc is here https://github.com/kodadot/nft-gallery/issues/7562#issuecomment-1772705044

Therefore you can ask api if collection was migrated or not

> Does it mean migration was impossible if the collection owner had yet to migrate? Is that correct?

Sorry, I do not understand your question. Can you please rephrase your request?

**Comment by preschian** [comment]
> Good Q.
> Will add it to the API.
> The full desc is here [#7562 (comment)](https://github.com/kodadot/nft-gallery/issues/7562#issuecomment-1772705044)

👍👍



> > Does it mean migration was impossible if the collection owner had yet to migrate? Is that correct?
> 
> Sorry, I do not understand your question. Can you please rephrase your request?



oops, sorry, I mean, if the collection owner has **not yet migrated**. for this section:

![image](https://github.com/kodadot/nft-gallery/assets/734428/914b4d71-b567-4a41-9177-78e740e49e3b)

**Comment by vikiival** [comment]
> oops, sorry, I mean, if the collection owner has **not yet migrated**. for this section:

Like You can migrate anything 

API has call for all you your collection that you have migrated

```
GET /relocations/issuers/${accountId}
```

returns a list

![Screenshot 2023-10-25 at 17 13 26](https://github.com/kodadot/nft-gallery/assets/22471030/3743bde2-23b0-49d2-9be8-fc446c22d2ec)

**Comment by preschian** [comment]
> API has call for all you your collection that you have migrated
> 
> ```
> GET /relocations/issuers/${accountId}
> ```
> 
> returns a list
> 
> Currently I am unable to deploy waifu worker

lmk if the endpoint deployed 👍 I will open followup PR



> Like You can migrate anything

@vikiival @exezbcz, should we update the "Migration Not Possible" design?

**Comment by vikiival** [comment]
> mk if the endpoint deployed 👍 I will open followup PR

published the **waifu** worker

**Comment by exezbcz** [comment]
> should we update the "Migration Not Possible" design?

yeah we can remove it :D

**Comment by vikiival** [comment]
cc @preschian any blocker here?

**Comment by preschian** [comment]
> any blocker here?

hi hi, I need help with this one #8206

regarding this https://github.com/kodadot/nft-gallery/pull/8201#issuecomment-1824594927, iiuc @exezbcz will have some updated version

**Comment by exezbcz** [comment]
> > should we update the "Migration Not Possible" design?
> 
> yeah we can remove it :D

I guess this, the waiting for your action design is still valid

**Comment by preschian** [comment]
> > > should we update the "Migration Not Possible" design?
> > 
> > 
> > yeah we can remove it :D
> 
> I guess this, the waiting for your action design is still valid

@exezbcz hhmm, I don't understand. Could you elaborate more? In PR #8201, that section was removed. need to adjust this one, right? https://github.com/kodadot/nft-gallery/pull/8201#issuecomment-1824594927

**Comment by exezbcz** [comment]
> * `for the bottom part, I will probably change the text a bit` -> what is the text?

this will be removed, the part is not needed




> * what/how is the correct one?

correct one is that it shows what is described in the issue 

>- here are items from collections you don't own, but the collection was migrated and initiated - giving you the option to migrate as well.
>- Displayed here are individual items that you own from collections that have been migrated. Ensure you're up-to-date by migrating these pieces right away. Please note you can only start the migration once the collection owner initiates it.



does that make sense? cc @preschian

**Comment by preschian** [comment]
@exezbcz I probably misunderstood your requirements. Here are some updates in #8280

"Budget Spaceships" -> I have one item from this collection. But, the owner still has not migrated the collection yet -> there are no buttons on the card.
"AI Cyberpunks" -> I have one item, and the owner already migrated that to ahk -> there is a button to migrate my items.

Is that correct?

![image](https://github.com/kodadot/nft-gallery/assets/734428/70ac131e-2e24-4795-8e6d-42155f2eeaf2)

**Comment by preschian** [comment]
@exezbcz I mocked the address locally with your address. It should show up like this

![image](https://github.com/kodadot/nft-gallery/assets/734428/74a07417-cbb1-4090-af6a-960cc68743dc)

**Comment by preschian** [comment]
hmmm, wait, maybe the titles between "Ready for Migration" and "Waiting For Your Action" are confusing. We need to change the words, I guess? Because the user can't do anything unless the collection owner migrates the collection

**Comment by exezbcz** [comment]
> Budget Spaceships" -> I have one item from this collection. But, the owner still has not migrated the collection yet -> there are no buttons on the card.

the card should not be visible, its showed only after its possible to follow the migration, that is why its "waiting for your action" 




> It should show up like this

yeah but without the waiting for your action section - owner of the collections did not migrate the collection yet

**Comment by preschian** [comment]
> the card should not be visible

aahh, I see, hide the card completely instead. will update that

**Comment by exezbcz** [comment]
@preschian correct, plus when there is no card, the section is hidden
